src/applications/caregivers/utils/helpers.js

```diff
@@ -1,150 +1,112 @@
-import { isEmpty, mapValues } from 'lodash';
-import caregiverFacilities from 'vets-json-schema/dist/caregiverProgramFacilities.json';
+import { camelCase, isEmpty, omit } from 'lodash';
 import { transformForSubmit } from 'platform/forms-system/src/js/helpers';
-import {
-  primaryCaregiverFields,
-  secondaryOneFields,
-} from '../definitions/constants';
-
-// Merges all the state facilities into one object with values as keys
-// and labels as values
-export const medicalCenterLabels = Object.keys(caregiverFacilities).reduce(
-  (labels, state) => {
-    const stateLabels = caregiverFacilities[state].reduce(
-      (centers, center) =>
-        Object.assign(centers, {
-          [center.code]: center.label,
-        }),
-      {},
-    );
-
-    return Object.assign(labels, stateLabels);
-  },
-  {},
-);
-
-// Turns the facility list for each state into an array of strings
-export const medicalCentersByState = mapValues(caregiverFacilities, val =>
-  val.map(center => center.code),
-);
-
-// transforms forData to match fullSchema structure for backend submission
-export const submitTransform = (formConfig, form) => {
-  const hasPrimary = form.data['view:hasPrimaryCaregiver'] ? 'primary' : null;
 
-  const hasSecondaryOne = form.data['view:hasSecondaryCaregiverOne']
+// transforms form data to match fullSchema structure for backend submission
+export const submitTransform = (formConfig, form) => {
+  const { data: formData } = form;
+  const hasPrimary = formData['view:hasPrimaryCaregiver'] ? 'primary' : null;
+  const hasSecondaryOne = formData['view:hasSecondaryCaregiverOne']
     ? 'secondaryOne'
     : null;
-
-  const hasSecondaryTwo = form.data['view:hasSecondaryCaregiverTwo']
+  const hasSecondaryTwo = formData['view:hasSecondaryCaregiverTwo']
     ? 'secondaryTwo'
     : null;
 
   // creates chapter objects by matching chapter prefixes
-  const buildChapterSortedObject = (data, dataPrefix) => {
+  const buildChapterSortedObject = dataPrefix => {
     // check to make sure there is a keyName
     if (dataPrefix === null) return {};
-    const keys = Object.keys(data);
 
     // matches prefix to fullSchema chapter object labels/keys
     const getChapterName = key => {
-      switch (key) {
-        case 'veteran':
-          return 'veteran';
-        case 'primary':
-          return 'primaryCaregiver';
-        case 'secondaryOne':
-          return 'secondaryCaregiverOne';
-        case 'secondaryTwo':
-          return 'secondaryCaregiverTwo';
-        default:
-          return null;
-      }
+      const keyMap = {
+        veteran: 'veteran',
+        primary: 'primaryCaregiver',
+        secondaryOne: 'secondaryCaregiverOne',
+        secondaryTwo: 'secondaryCaregiverTwo',
+      };
+      return keyMap[key] || null;
     };
 
+    const dataKeys = Object.keys(formData).filter(k => k.includes(dataPrefix));
```

Comment:
We don't need to map over all data keys. We only care about the ones that pertain to the data prefix passed into this method.